account_asset_management/models/account_move.py

```diff
@@ -30,13 +30,18 @@ class AccountMove(models.Model):
     asset_count = fields.Integer(compute="_compute_asset_count")
 
     def _compute_asset_count(self):
-        for rec in self:
-            assets = (
-                self.env["account.asset.line"]
-                .search([("move_id", "=", self.id)])
-                .mapped("asset_id")
+        if len(self) < 100:
+            for move in self:
+                move.asset_count = len(move.line_ids.filtered("asset_id"))
+        else:
+            self.asset_count = 0
+            asset_groups = self.env["account.move.line"].read_group(
+                [("move_id", "in", self.ids)], ["move_id", "asset_id"], ["move_id"]
             )
-            rec.asset_count = len(assets)
+            for asset_group in asset_groups:
+                self.browse(asset_group["move_id"][0]).asset_count = asset_group[
+                    "move_id_count"
+                ]
```

Comment:
Re-doing a browse can cause conflicts with the cache. Please do instead:
```suggestion
            vals = {x["move_id"][0]: x["move_id_count"] for x in asset_groups}
            for move in self:
                self.asset_count = vals.get(self.id, 0)
```
and no need to do the initial assignment. For the record, this also serves for virtual records not yet created in DB.

Reply:
What you suggest is the way I normally do it, but I wanted try the cache improvements in v13.
In what cases can it be conflictive?

Reply:
The idea of this is that if we have 1 million moves, but only 100 have related assets, the loop is 100 interactions instead of 1,000,000.

Reply:
It seems that from Odoo they also trust in the v13 cache:
https://github.com/odoo/odoo/blob/13.0/addons/stock_account/models/product.py#L114

Reply:
Well, I wouldn't take that code as reference, as there's an `|=` later, which recreates on each loop a new set (that's how it's implemented), which is not very performant.